Leetcode/aoc_2015_d6.py

```diff
@@ -0,0 +1,161 @@
+import unittest
+'''
+ ----- Part 1 -----
+It's an interesting problem, I personally have a hard time visualizing coordinate grids
+like this problem. Given a coordinate grid, 0,0 being top left, 999,0 being top right,
+0,999 being bottom left, and 999,999 being bottom right.
+Given instructions, perform actions. e.x:
+turn off xxx,yyy through xxx,yyy
+turn on 0,0 through 999,999 turns on every single light.
+toggle 0,0 through 999,0 toggles the first line of 1000 lights, meaning if they're off, they get switched on, off if
+they were already on.
+turn off, self explantatory.
+Return value is how many lights remain on.
+
+All lights are off by default.
+Make a count variable that tracks how many lights are on. Add one if turning on, subtract one when turning off.
+Do this while taking in the instructions so that it won't need to parse the list after all the instructions to count
+the amount of lights on.
```

Comment:
This is cute, but I'm personally not a fan. Yes, it lets you avoid one iteration of looping the array, but that's very much a micro-optimization. When you track this as a separate variable, you're increasing the surface area of the code, which increases how many potential places you could introduce a bug.

If you write the function to count the total brightness, that's fairly straightfoward code to write - and at least then if you know you got the wrong answer, it's an issue with how you're manipulating the brightness, not a bug that showed up because you're not counting correctly.

Reply:
That makes sense, funny actually, because I ran into a bug and couldn't figure out what was going on at first.
And at one point I thought maybe it was with the counter (it wasn't).